### PR TITLE
first version of search

### PR DESCRIPTION
**Pull request by ardan-bkennedy**
This is using the feed mechanics right now. I would like a readability and engineering review. Once that is done, I will focus on replacing the feed system for hitting the sales-api service.

### CODE REVIEW

File: cmd/search/views/views.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	// Generate the markup for the search template.
+	markup.Reset()
+	if err := views[SEARCH].Execute(&markup, fv); err != nil {
+		return nil, fmt.Errorf("error processing search template : %v", err)
+	}
+
+	// Generate the final markup with the layout template.
+	vars := map[string]interface{}{"LayoutContent": template.HTML(markup.String())}
+	markup.Reset()
+	if err := views[LAYOUT].Execute(&markup, vars); err != nil {
```
Comment (jcbwlkr) [inline cmd/search/views/views.go]: `html/template` provides support for nesting blocks of templates together so we only need one call to `*template.Execute`. I will look into this but it doesn't need to hold up the work on this PR.

File: cmd/search/views/views.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	return markup.Bytes(), nil
+}
+
+// Init loads the existing templates for use to generate views.
+func Init() error {
+
+	// In order for the endpoint tests to run this needs to be
+	// physically located. Trying to avoid configuration for now.
+	pwd, _ := os.Getwd()
+	if err := loadTemplate(LAYOUT, pwd+"/views/basic_layout.html"); err != nil {
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply cmd/search/views/views.go]: I haven't though this out yet outside of knowing I can copy this where I want it in the container. I like your idea of embedding this into the binary.

File: cmd/search/internal/search/search.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			if err != nil {
+
+				// Important to capture there errors.
+				v, ok := ctx.Value(web.KeyValues).(*web.Values)
+				if !ok {
+					log.Printf("TRACEID MISSING : search : ERROR : %s : searching feed : %v", searcher.FeedName(), err)
+				} else {
+					log.Printf("%s : search : ERROR : %s : search feed : %v", v.TraceID, searcher.FeedName(), err)
+				}
+			}
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply cmd/search/internal/search/search.go]: Interesting idea. I will try to implement it and see how it goes!!

File: cmd/search/main.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	// Logging
+
+	log := log.New(os.Stdout, "SEARCH : ", log.LstdFlags|log.Lmicroseconds|log.Lshortfile)
+
+	// =========================================================================
+	// Configuration
+
+	var cfg struct {
+		Web struct {
+			APIHost         string        `conf:"default:0.0.0.0:5000"`
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply cmd/search/main.go]: true

File: cmd/search/views/views.go (hunk 1 above)
Comment (ardan-bkennedy) [reply cmd/search/views/views.go]: Please do, that would be amazing.

File: cmd/search/handlers/routes.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/ardanlabs/service/internal/mid"
+	"github.com/ardanlabs/service/internal/platform/web"
+)
+
+// API returns a handler for a set of routes.
+func API(shutdown chan os.Signal, log *log.Logger) http.Handler {
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply cmd/search/handlers/routes.go]: We should use the Routes name in all the projects?

**Pull request closed without merging**